marketplace-service/src/main/java/com/axonivy/market/service/impl/VersionServiceImpl.java

```diff
@@ -52,62 +41,69 @@
 
 @Log4j2
 @Service
+@AllArgsConstructor
 public class VersionServiceImpl implements VersionService {
 
-  private final GHAxonIvyProductRepoService gitHubService;
   private final MavenArtifactVersionRepository mavenArtifactVersionRepository;
   private final ProductRepository productRepo;
   private final ProductJsonContentRepository productJsonRepo;
   private final ProductModuleContentRepository productContentRepo;
+  private final ProductJsonContentService productJsonContentService;
   private final ObjectMapper mapper = new ObjectMapper();
 
-  public VersionServiceImpl(GHAxonIvyProductRepoService gitHubService,
-      MavenArtifactVersionRepository mavenArtifactVersionRepository, ProductRepository productRepo,
-      ProductJsonContentRepository productJsonRepo, ProductModuleContentRepository productContentRepo) {
-    this.gitHubService = gitHubService;
-    this.mavenArtifactVersionRepository = mavenArtifactVersionRepository;
-    this.productRepo = productRepo;
-    this.productJsonRepo = productJsonRepo;
-    this.productContentRepo = productContentRepo;
-  }
-
-  public static Map<String, List<ArchivedArtifact>> getArchivedArtifactMapFromProduct(
-      List<MavenArtifact> artifactsFromMeta) {
-    Map<String, List<ArchivedArtifact>> result = new HashMap<>();
-    artifactsFromMeta.forEach(artifact -> {
-      List<ArchivedArtifact> archivedArtifacts = new ArrayList<>(
-          Optional.ofNullable(artifact.getArchivedArtifacts()).orElse(Collections.emptyList()).stream()
-              .sorted(new ArchivedArtifactsComparator()).toList());
-      Collections.reverse(archivedArtifacts);
-      result.put(artifact.getArtifactId(), archivedArtifacts);
-    });
-    return result;
-  }
-
   public List<MavenArtifactVersionModel> getArtifactsAndVersionToDisplay(String productId, Boolean isShowDevVersion,
       String designerVersion) {
-    List<String> versionsToDisplay = VersionUtils.getVersionsToDisplay(getPersistedVersions(productId),
+    MavenArtifactVersion cache = mavenArtifactVersionRepository.findById(productId).orElse(
+        MavenArtifactVersion.builder().productId(productId).build());
+    List<String> versionsToDisplay = VersionUtils.getVersionsToDisplay(new ArrayList<>(cache.getProductArtifactsByVersion().keySet()),
```

Comment:
cache => is not mean full for readable => can we rename it to existingMavenArtifactVersion or something like that?
and maybe NPE can happen cache.getProductArtifactsByVersion().keySet() if .orElse() of cache is used